anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/rcf/RandomCutForestAnomalyDetector.java

```diff
@@ -109,29 +132,35 @@ public AnomalyResult classify(MetricData metricData) {
     }
 
     /**
-     * Invokes the endpoint with the data that is held in the Shingle queue.
-     * @return AWS anomaly score as a double
+     * Invokes the endpoint with the data that is held in the Shingle queue. This version supports sending one shingle
+     * at a time to predict endpoint and getting a single score back.
+     **
+     * @return AWS anomaly score(s) as a double
      */
     private double getAnomalyScore() {
+        // TODO Note: Future versions can easily support sending multiple shingles in one request and getting multiple
+        // scores back by simple updates to this class. [TK]
         final String shingleBody = this.shingle.toCsv().get();
         final Optional<ByteBuffer> bodyBuffer =
                 Optional.of(ByteBuffer.wrap(shingleBody.getBytes(StandardCharsets.UTF_8)));
 
         if (bodyBuffer.isPresent()) {
-            invokeEndpointRequest.setBody(bodyBuffer.get());
-            invokeEndpointRequest.setEndpointName(ENDPOINT);
-            invokeEndpointRequest.setAccept(APPLICATION_JSON_ACCEPT);
+            this.invokeEndpointRequest.setBody(bodyBuffer.get());
```

Comment:
Is it ever possible for this to be called concurrently? Worried about reusing the same request.